### PR TITLE
Scripts

### LINKED ISSUE #1: Unify esm scripts
Use a shared repository for access-esm payu scripts, maybe a git subtree

### PR DESCRIPTION
**Pull request by ScottWales**
General script cleanup

 - **scripts/** directory now comes from coecms/esm-scripts - still needs to be synced once this is merged, fixes #1 
 - **warm-start.sh** split into two parts
    - top level script sets the run to copy from, this will vary between different models
    - version in scripts/ directory common to all runs
    - reads the run start date from payu config.yaml, sets the run start date in the atmosphere namelist
 - **pre.sh** script takes the land use file as an argument, as this will vary between different runs

Fixes to make bit-reproducible with HI-24

 - Ocean horizontal decomposition changed
 - Fixes to ice initial date settings
 - Fixes to ice restart file (payu overwrites the path in **ice.restart_file** I had set, so the ice restart file has been renamed so payu picks it up automatically)
 - Change 'pre' script so that it uses the land use of the current year, rather than current year + 1
 - Change 'pre' script to no-op if there is no land use data for the current year (since land use starts at 1851)

### CODE REVIEW

**Comment by ScottWales** [comment]
@HoWol76 can you take a look over the new scripts please

File: config.yaml [hunk 1]
```diff
@@ -19,7 +19,8 @@ submodels:
 
     - name: ocean
       model: mom
-      ncpus: 228
+      #ncpus: 228
```
Comment (aidanheerdegen) [inline config.yaml]: If this is a permanent change will want to remove this commented line

File: ocean/input.nml [hunk 2]
```diff
@@ -197,7 +197,8 @@
 /
 
  &ocean_model_nml
-      layout = 19,12
+      #layout = 19,12
```
Comment (aidanheerdegen) [inline ocean/input.nml]: Ditto for this

File: config.yaml (hunk 1 above)
Comment (aidanheerdegen) [reply config.yaml]: Also should remove the commented walltime option and the project also.

Comment (ScottWales) [reply config.yaml]: ta, done

**Pull request merged**